frontend/src/metabase-types/api/mocks/dataset.ts

```diff
@@ -25,6 +25,16 @@ export const createMockColumn = (
   };
 };
 
+export const createMockCustomColumn = (
+  data: Partial<DatasetColumn> = {},
+): DatasetColumn => {
+  return createMockColumn({
+    id: undefined,
+    table_id: undefined,
+    ...data,
+  });
+};
```

Comment:
I'm worried it's still possible to create a field with that helper — TypeScript would allow `id` and `table_id` and `...data` spreading will overwrite the default `undefined` values

```ts
createMockCustomColumn({ id: 10 })
```